### PR TITLE
Add simple message view

### PR DESCRIPTION
**Pull request by greyrhino**

### CODE REVIEW

**Comment by jfindley** [comment]
We can't use a global map for this as it's not thread-safe.

**Comment by greyrhino** [comment]
Have updated with urlParams as a field on api.

File: secrets/secrets.go [hunk 1]
```diff
@@ -3,14 +3,16 @@ package secrets
...
+
 	"github.com/pborman/uuid"
 	"golang.org/x/crypto/curve25519"
 	"golang.org/x/crypto/nacl/box"
 	"golang.org/x/crypto/nacl/secretbox"
-	"io"
 )
 
 var master *[32]byte
+var MasterKeyName = "master"
```
Comment (jfindley) [inline secrets/secrets.go]: Should this be a const?

File: main.go [hunk 2]
```diff
@@ -13,8 +14,15 @@ import (
...
 var database db.DB
 
+var secretIDRegex *regexp.Regexp
+var secretKeyRegex *regexp.Regexp
+
 func init() {
 	database = new(postgres.DB)
+
+	// Compile credential checking regex patterns.
+	secretIDRegex = regexp.MustCompile(`^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`)
```
Comment (jfindley) [inline main.go]: Might be nicer in api.go?

File: main.go [hunk 3]
```diff
@@ -13,8 +14,15 @@ import (
 
 var database db.DB
 
+var secretIDRegex *regexp.Regexp
+var secretKeyRegex *regexp.Regexp
```
Comment (jfindley) [inline main.go]: Move to api.go?

**Pull request merged**